Tools/clinic/clinic.py

```diff
@@ -2652,7 +2652,7 @@ def __init__(self, name, py_name, function, default=unspecified, *, c_default=No
         self.name = ensure_legal_c_identifier(name)
         self.py_name = py_name
 
-        if default is not unspecified:
+        if default is not unspecified and (default is not NULL or c_default == 'NULL'):
```

Comment:
What does this change do? Is it related to the signature?

Reply:
Seems it is a leftover from my previous attempts. It does not affect current output, although there may be cases in which it makes difference. But for now I remove this change.